src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -0,0 +1,180 @@
+extern crate bazel_protos;
+extern crate clap;
+extern crate fs;
+
+use clap::{App, Arg, SubCommand};
+use fs::hash::Fingerprint;
+use fs::store::Store;
+use std::error::Error;
+use std::fs::File;
+use std::io::{self, Read, Write};
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+use std::process::exit;
+
+fn main() {
+  match execute(
+    App::new("fs_util")
+      .subcommand(
+        SubCommand::with_name("file")
+          .subcommand(
+            SubCommand::with_name("cat")
+              .about("Output the contents of a file by fingerprint.")
+              .arg(Arg::with_name("fingerprint").required(true).takes_value(
+                true,
+              )),
+          )
+          .subcommand(
+            SubCommand::with_name("save")
+              .about(
+                "Ingest a file by path, which allows it to be used in Directories/Snapshots. \
+Outputs a fingerprint of its contents.",
+              )
+              .arg(Arg::with_name("path").required(true).takes_value(true)),
+          ),
+      )
+      .subcommand(
+        SubCommand::with_name("directory").subcommand(
+          SubCommand::with_name("save")
+            .about(
+              "Ingest a directory recursively. Saves all files found therein and saves Directory \
+protos for each directory found. Outputs a fingerprint of the canonical top-level Directory proto.",
+            )
+            .arg(Arg::with_name("source").required(true).takes_value(true)),
+        ),
+      )
+      .arg(
+        Arg::with_name("local_store_path")
+            .takes_value(true)
+            // TODO: Default this to wherever pants actually stores this.
```

Comment:
Maybe a good idea to make this required to improve discoverability until we fix that.

Reply:
Done